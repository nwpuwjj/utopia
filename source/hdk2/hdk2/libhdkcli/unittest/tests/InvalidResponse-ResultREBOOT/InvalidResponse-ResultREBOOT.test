#
# InvalidResponse-ResultREBOOT.test
#   Test the handling of a response with an HNAP REBOOT result, but not all required elements.
#

method: http://cisco.com/Unittest/ComplexUnittestMethod
host: http://127.0.0.1:8080
