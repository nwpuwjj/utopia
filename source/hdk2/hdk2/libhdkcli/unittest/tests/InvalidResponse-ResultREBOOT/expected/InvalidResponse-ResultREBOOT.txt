+++++++ Request Headers +++++++
HTTP/1.1 POST /HNAP1
Connection: Close
Content-Type: text/xml; charset="utf-8"
Content-Length: 1002
Authorization: Basic Og==
SOAPAction: "http://cisco.com/Unittest/ComplexUnittestMethod"
Host: 127.0.0.1:8080

+++++++ Request Data +++++++
<?xml version="1.0" encoding="utf-8"?>
<Envelope xmlns="http://schemas.xmlsoap.org/soap/envelope/">
<Body>
<ComplexUnittestMethod xmlns="http://cisco.com/Unittest/">
<StructInputArg>
<StringMember>P___ C______ (feat. P___ B_____)</StringMember>
<IntMember>0</IntMember>
<BoolMember>false</BoolMember>
<DateTimeMember>1985-05-27T00:00:01Z</DateTimeMember>
<IPAddressMember>127.0.0.1</IPAddressMember>
<MACAddressMember>04:08:15:16:23:42</MACAddressMember>
<EnumMember>EnumValue1</EnumMember>
<StringArrayMember>
<string>Better forget it ....</string>
<string>You&apos;ll never get it...</string>
<string>Oh she&apos;ll say there&apos;s no other...</string>
<string>.. until she finds another...</string>
<string>...</string>
<string>...</string>
<string>She&apos;s an easy lover</string>
<string>She&apos;ll get a hold on you believe it!</string>
</StringArrayMember>
</StructInputArg>
<StringInputArg>????</StringInputArg>
<BoolInputArg>true</BoolInputArg>
</ComplexUnittestMethod>
</Body>
</Envelope>


------- Response -------
HTTP/1.1 200 OK
Content-Type: text/xml; charset=utf-8
Connection: close

<?xml version="1.0" encoding="utf-8" ?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
<soap:Body>
  <ComplexUnittestMethodResponse xmlns="http://cisco.com/Unittest/">
      <ComplexUnittestMethodResult>REBOOT</ComplexUnittestMethodResult>
      <!-- <IntOutputArg>223344</IntOutputArg> -->
      <StringOutArg>The IntOutputArg is missing from this response</StringOutArg>
  </ComplexUnittestMethodResponse>
</soap:Body>
</soap:Envelope>

******* Client Result ******

      SOAP action: http://cisco.com/Unittest/ComplexUnittestMethod
        HTTP host: http://127.0.0.1:8080
Network Object ID: (null)
    HTTP username: (null)
    HTTP password: (null)
          timeout: 0
       HNAP input: InvalidResponse-ResultREBOOT.input.xml

HNAP call failed with error ResponseInvalid (2)
malloc_interposer.c - ************************************************************
malloc_interposer.c - exit: malloc called 116 times
malloc_interposer.c - exit: free called 116 times
malloc_interposer.c - exit: malloc current = 0
malloc_interposer.c - exit: malloc total = 21168
malloc_interposer.c - exit: malloc max = 12398
malloc_interposer.c - ************************************************************
